cmd/activator/main.go

```diff
@@ -162,29 +161,21 @@ func main() {
 	// Enable TLS against queue-proxy when internal-encryption is enabled.
 	tlsEnabled := networkConfig.InternalEncryption
 
+	var certCache *certificate.CertCache
+	if tlsEnabled {
+		logger.Info("Internal Encryption is enabled")
+		certCache = certificate.NewCertCache(ctx)
+	}
+
 	// Enable TLS client when queue-proxy-ca is specified.
 	// At this moment activator with TLS does not disable HTTP.
 	// See also https://github.com/knative/serving/issues/12808.
 	if tlsEnabled {
-		logger.Info("Internal Encryption is enabled")
-		caSecret, err := kubeClient.CoreV1().Secrets(system.Namespace()).Get(ctx, netcfg.ServingInternalCertName, metav1.GetOptions{})
-		if err != nil {
-			logger.Fatalw("Failed to get secret", zap.Error(err))
-		}
-
-		pool, err := x509.SystemCertPool()
-		if err != nil {
-			pool = x509.NewCertPool()
-		}
-
-		if ok := pool.AppendCertsFromPEM(caSecret.Data[certificates.CaCertName]); !ok {
-			logger.Fatalw("Failed to append ca cert to the RootCAs")
-		}
-
 		tlsConf := &tls.Config{
-			RootCAs:            pool,
-			InsecureSkipVerify: false,
-			ServerName:         certificates.FakeDnsName,
+			VerifyPeerCertificate: certCache.ValidateCert,
```

Comment:
1.  "_VerifyPeerCertificate, if not nil, is called after normal certificate verification by either a TLS client or server._"  - so to use this function we need to make sure standard verification works, this means we need to also update **RootCAs**
2. Best to avoid replacing the golang tls lib verification with our own unless we are compelled to do that. Better to try to use the library tls verification (and any future updates it has) and add verifications as needed.
3. Can we find a way to update tls.Config with the changes on every connection attempt made (or find a way to update the used tls.Config on any change we detect?)   
4. Instead of (3), can we change our install to ensure certificates are created before we get started and have the certificates created regardless if encryption is used, then require that the activator is restarted when a CA changes?